third-party/nix/release.nix

```diff
@@ -3,7 +3,7 @@
 
 let
   pkgs = import <nixpkgs> { overlays = [ moz_overlay ]; };
-  channel = pkgs.rustChannelOf { date = "2022-08-05"; channel = "nightly"; };
+  channel = pkgs.rustChannelOf { date = "2023-01-10"; channel = "nightly"; };
```

Comment:
I believe if you just put it like this:

`pkgs.rustChannelOf { };`
it will pick up the one defined in the toolchain file.

Not sure if there's a nicer way to define this - perhaps removing `channel` and `rust-nightly` altogether?

Reply:
Maybe this will help?
https://github.com/mozilla/nixpkgs-mozilla#using-in-nix-expressions
```
(nixpkgs.rustChannelOf { rustToolchain = ./rust-toolchain.toml; }).rust
```

Reply:
If it works, sure.

Let's just avoid having to maintain this file.
(we don't use `nix` in CI anymore)

Reply:
@ashutoshvarma can you apply your suggested change? :slightly_smiling_face:

Reply:
Well, that didn't work unfortunately, https://github.com/mozilla/nixpkgs-mozilla/issues/287
I've switched moz-overlay with rust-overlay and it works!